lib/transforms/onnxextension_legalizer.cc

```diff
@@ -738,6 +738,34 @@ static std::vector<Def> ConvertSplit(const ONNXExtensionInst* ext,
   return ret_v;
 }
 
+std::vector<Def> ConvertGlobalMaxPooling(const ONNXExtensionInst* ext,
+                                         IRBuilder* builder) {
+  HLCHECK(ext->GetNumOfOperands() == 1);
+  auto input = ext->GetOperand(0);
+  const auto& input_type = input.GetType();
```

Comment:
check if input type is valid ? like "input_type.IsValid()"?

Reply:
Even for valid type, we can also add a HLCHECK to make sure dims >= 4

Reply:
right